src/lib.rs

```diff
@@ -78,32 +74,36 @@ pub fn parse_line<S: AsRef<str>>(line: S) -> Option<TestResult> {
 /// parse_line parses a line
 /// ```
 pub fn prettify<S: AsRef<str>>(input: S) -> String {
-    let mut output = String::new();
     if let Some((fn_name, sentence)) = input.as_ref().split_once("_fn_") {
-        output.push_str(fn_name);
-        output.push(' ');
-        output.push_str(sentence.replace('_', " ").as_ref());
+        format!("{} {}", fn_name, humanize(sentence))
     } else {
-        output.push_str(input.as_ref().replace('_', " ").as_ref());
+        humanize(input)
     }
-    output
+}
+
+fn humanize<S: AsRef<str>>(input: S) -> String {
```

Comment:
Does `input: impl AsRef<str>` work here? If so, it's always nice to eliminate a generic.

Reply:
It appears to work! Should we make a similar change to the `prettify` function?